src/ploomber/dag/plot.py

```diff
@@ -71,27 +72,28 @@ def with_d3(graph, output, image_only=False):
         Path(output).write_text(rendered)
 
 
-@requires(['requests_html', 'nest_asyncio'],
-          name='embedded HTML with D3 backend',
-          pip_names=['requests-html', 'nest_asyncio'])
 def embedded_html(path):
-    import asyncio
-
-    import nest_asyncio
-    nest_asyncio.apply()
-    return asyncio.get_event_loop().run_until_complete(
-        _embedded_html(path=path))
-
-
-async def _embedded_html(path):
-    # https://github.com/jupyter/nbclient/blob/1d629b2bed561fde521e6408e190a8159f117ddc/nbclient/util.py
-    # https://github.com/jupyter/nbclient/blob/main/requirements.txt
-    from requests_html import HTML as HTML_
-    from requests_html import AsyncHTMLSession
-
-    session = AsyncHTMLSession()
-    html = HTML_(html=Path(path).read_text(), session=session)
-    await html.arender()
-    # ensure we close the session, otherwise this will fail on windows
-    await session.close()
-    return HTML(data=html.html)
+    # set output cell to hold 100% of the embedded content
+    display(HTML('<style>.output.output_scroll { height: 100% }</style>'))
+
+    # create a temp copy of the file to embed in a local
+    # directory inorder to load it with iFrame
+    embedded_assets_dir = 'tmp-embedded-assets'
+    isTempEmbeddedAssetsDirExists = os.path.isdir(embedded_assets_dir)
+
+    if isTempEmbeddedAssetsDirExists:
```

Comment:
Agree. For a cleaner code, I've replaced this logic with `clear_embedded_assets_dir` method and changed `tmp-embedded-assets` to `embedded-assts` since it's not a temp directory (for this reason I didn't use `tempfiles`).

Reply:
It's a good point and makes more sense.

I've talked about this with Eduardo [here](https://github.com/ploomber/ploomber/pull/986#discussion_r962332327).